CRM/Member/Form/Search.php

```diff
@@ -305,8 +305,13 @@ public function fixFormValues() {
     $membershipType = CRM_Utils_Request::retrieve('type', 'String');
 
     if ($membershipType) {
-      $this->_formValues['membership_type_id'] = array($membershipType);
-      $this->_defaults['membership_type_id'] = array($membershipType);
+			$typearray=explode(",",$membershipType);
+			$typeresult = array();
+			foreach ($typearray as $type){
+				$typeresult+=array($type => 1);
+			}
```

Comment:
This could be written more concisely as `array_fill_values(explode(',', $membershipType), 1);`

But I'm confused. The existing code produced an array like `[0 => $membershipType]` and your new code produces `[$membershipType => 1]`. Which is it?